code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -354,3 +354,27 @@
 /mob/living/simple_animal/drone/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0)
 	if(affect_silicon)
 		return ..()
+
+/mob/living/silicon/robot/drone/decompile_act(obj/item/matter_decompiler/C, mob/user)
+	if(!client && istype(user, /mob/living/silicon/robot/drone))
+		var/mob/living/silicon/robot/drone/D = C.loc
+		to_chat(D, "<span class='warning'>You begin decompiling the other drone.</span>")
+
+		if(!do_after(D, 50, target = src.loc))
+			to_chat(D, "<span class='warning'>You need to remain still while decompiling such a large object.</span>")
+			return
+
+		if(!src || !D) return ..()
```

Comment:
I don't think checking for `!src` is necessary here?

Reply:
It can be possible that src is being qdeleted and it fails somehow. Yet `!src` won't catch that.
`QDELETED(src)` will catch that.
Also put the `return ..()` on a new line